### ISSUE TITLE
pt-verify-checksum reports a lot of errors if checksums are disabled

### ISSUE THREAD

**Issue by mbanck** [uptimejp/postgres-toolkit#44]
`pt-verify-checksum` does not seem to check whether checksums are actually enabled, and if they are not, it reports an error for each block.

**Comment by mbanck** [comment]
Patch for this, in `verifychecksum.c`:

```
--- verifychecksum.c.orig       2018-01-23 14:12:23.000000000 +0100
+++ verifychecksum.c.new        2018-01-23 14:13:56.000000000 +0100
@@ -13,6 +13,14 @@
 int verbose = 0;
 
 int
+detect_checksum(const char *page)
+{
+  PageHeader   phdr = (PageHeader) page;
+
+  return (phdr->pd_checksum != 0);
+}
+
+int
 verify_page(const char *page, BlockNumber blkno, const char *filepath)
 {
   PageHeader   phdr = (PageHeader) page;
@@ -47,6 +55,12 @@
 
   while (read(fd, page, BLCKSZ) == BLCKSZ)
     {
+      if (blkno == 0 && detect_checksum(page) == 0)
+       {
+         /* No checksums to verify. */
+         return TRUE;
+       }
+
       if (!verify_page(page, blkno, filepath))
        {
          corrupted++;
```

**Comment by mbanck** [comment]
Better version of the `detect_checksum()` function:

```
--- verifychecksum.c.orig       2018-01-23 14:12:23.000000000 +0100
+++ verifychecksum.c    2018-01-23 15:01:50.000000000 +0100
@@ -13,6 +14,19 @@
 int verbose = 0;

 int
+detect_checksum(const char *page)
+{
+  PageHeader   phdr = (PageHeader) page;
+
+  /*
+   * phdr->pd_checksum is 0 if no checksums are enabled and 1 if checksums are
+   * not supported (i.e. 9.2 and lower). Consider everything else as checksums
+   * present.
+   */
+  return (phdr->pd_checksum < 0 || phdr->pd_checksum > 1);
+}
+
+int
 verify_page(const char *page, BlockNumber blkno, const char *filepath)
 {
   PageHeader   phdr = (PageHeader) page;
```

**Comment by snaga** [comment]
Commited a fix at 404e4fe.

**Comment by snaga** [comment]
I modified verifychecksum to exit with different exit code because I want to detect if the data checksums are not enabled and/or not supported.

**Comment by snaga** [comment]
Merged #53.
